students/thorn/lesson04/mailroom.py

```diff
@@ -97,12 +72,27 @@ def create_report():
     """
     print("Donor:                    | $    Total     |   Donations   | $   Average   |")
     print("-"*76)
-    for item in donors:
-        amt_total = float(sum(donors[item]))
-        num_total = int(len(donors[item]))
-        # Thousand separator as default. Careful with the space if we get some big donors.
-        print("{:<26}| ${:>14,.2f}|{:>15}| ${:>13,.2f}".format(item, amt_total, num_total, amt_total/num_total))
-    print('\n')
+    # List is sorted by donor amounts desc.  Create preset list to pass in 
+    # vars once the list is sorted.
+    line_in = "{:<26}| ${:>14,.2f}|{:>15}| ${:>13,.2f}\n"
+    lines = []
+    # Append relevant information to the list to be sorted.
+    # List comprehension for fun
+    lines = [[donor for donor, donations in donors.items()],
+             [sum(donations) for donor, donations in donors.items()],
+             [len(donations) for donor, donations in donor.items()],
+             [(sum(donations)/len(donations)) for donor, donations in donor.items()]]
```

Comment:
a regular (single) loop would be better here because you are iterating over `donor` 4 times when it could be just once